pkg/collector/corechecks/system/gpu/nvidia/jetson.go

```diff
@@ -0,0 +1,488 @@
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build !windows
+
+package nvidia
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"gopkg.in/yaml.v2"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	checkName = "jetson"
+
+	kb = 1024
+	mb = kb * 1024
+	gb = mb * 1024
+
+	// Indices of the regex in the 'regexes' variable below
+	regexRAMIdx       = 0
+	regexSwapCacheIdx = 1
+	regexIRamIdx      = 2
+
+	// Regex used to parse the GPU usage and frequency => e.g. EMC_FREQ 7%@408 GR3D_FREQ 0%@76
+	regexGpuUsageIdx = 3
+
+	// Regex used to parse the CPU usage section => e.g. CPU [2%@102,1%@102,0%@102,0%@102]
+	regexCPUUsageIdx = 4
+
+	// Regex used to parse the temperature information => e.g. thermal@41C
+	regexTemperatureIdx = 5
+
+	// Regex used to parse the voltage information => e.g. POM_5V_IN 900/943
+	regexVoltageIdx = 6
+
+	// Regex used to parse cpu and freq => e.g. 2%@102
+	regexCPUFreqIdx = 7
+
+	// Indices of the matched fields by the RAM regex
+	ramUsed          = 1
+	totalRAM         = 2
+	ramUnit          = 3
+	numFreeBlock     = 4
+	largestFreeBlock = 5
+	lfbUnit          = 6
+
+	// Indices of the matched fields by the Swap/Cache regex
+	swapUsed  = 1
+	totalSwap = 2
+	swapUnit  = 3
+	cached    = 4
+	cacheUnit = 5
+
+	// Indices of the matched fields by the Icache regex
+	iramUsed    = 1
+	iramTotal   = 2
+	iramUnit    = 3
+	iramLfb     = 4
+	iramLfbUnit = 5
+
+	// Indices of the matched fields by the GPU usage regex
+	emcPct  = 1
+	emcFreq = 2
+	gpuPct  = 3
+	gpuFreq = 4
+
+	// Indices of the matched fields by the voltage regex
+	voltageProbeName = 1
+	currentVoltage   = 2
+	averageVoltage   = 3
+
+	// Indices of the matched fields by the temperature regex
+	tempZone  = 1
+	tempValue = 2
+
+	// Indices of the matched fields by the CPU regex
+	cpuUsage = 1
+	cpuFreq  = 2
+)
+
+var regexes = [...]string{
+	// Group 1.	-> Used
+	// Group 2.	-> Total
+	// Group 3.	-> Unit
+	// Group 4.	-> Number of LFB
+	// Group 5.	-> LFB
+	// Group 6.	-> Unit
+	`RAM\s*(\d+)/(\d+)([kKmMgG][bB])\s*\(lfb\s*(\d+)x(\d+)([kKmMgG][bB])\)`,
+
+	// Group 1.	-> Used
+	// Group 2.	-> Total
+	// Group 3.	-> Unit
+	// Group 4. -> Cached
+	// Group 5. -> Unit
+	`SWAP\s*(\d+)\/(\d+)([kKmMgG][bB])\s*\(cached\s*(\d+)([kKmMgG][bB])\)`,
+
+	// Group 1.	-> Used
+	// Group 2.	-> Total
+	// Group 3.	-> Unit
+	// Group 4.	-> LFB
+	// Group 5.	-> Unit
+	`IRAM\s*(\d+)\/(\d+)([kKmMgG][bB])\s*\(lfb\s*(\d+)([kKmMgG][bB])\)`,
+
+	// Group 1.	-> EMC %
+	// Group 2.	-> EMC Freq (opt)
+	// Group 3.	-> GPU %
+	// Group 4.	-> GPU Freq (opt)
+	`EMC_FREQ\s*(\d+)%(?:@(\d+))?\s*GR3D_FREQ\s*(\d+)%(?:@(\d+))?`,
+
+	`CPU\s*\[((?:\d+%@\d+,?)+)\]`,
```

Comment:
Could you add a comment with the meaning of the capture group on this one (comma-separated list of CPU % @ CPU freq)?